### PR TITLE
Add script for checking header files with clang

### PR DESCRIPTION
**Pull request by stblr**
TODO:
- [x] Windows
- [x] -Wall -Wextra
- [x] Add to CI

### CODE REVIEW

File: clang_check.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+def check_headers(extension, flags):
+    source = ""
+    for path in Path("source").rglob("*." + extension):
+        source += "#include \""
+        source += str(path).removeprefix("source/")
+        source += "\"\n"
+
+    subprocess.run(
+        ["clang"] + flags + ["-", "-o", DEV_NULL],
```
Comment (terorie) [inline clang_check.py:41]: Would `stdout=subprocess.DEVNULL` help here?

Comment (riidefi) [reply clang_check.py]: Oh, good to know! Switched to that

Comment (stblr) [reply clang_check.py:45]: I reverted it because we want to ignore the generated .o file, not redirect the standard output

**Comment by terorie** [comment]
LGTM. Nit: We could merge clang-format (Lint stage) and clang-check. They kinda do the same thing.

**Comment by riidefi** [comment]
Possibly, yea. Though let's open up a follow-up issue for that; merging this now.

**Pull request merged**